Frameworks/CoreImage/CIImage.mm

```diff
@@ -73,90 +74,164 @@ @implementation CIImage
 /**
  @Status Interoperable
 */
-- (CIImage*)init {
++ (CIImage*)emptyImage {
+    return [[[CIImage alloc] init] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (CGRect)extent {
+    return self->_extent;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)init {
     if (self = [super init]) {
-        _cgImage = nil;
-        _color = nil;
-        _filter = nil;
         _extent = CGRectInfinite;
     }
     return self;
 }
 
+- (instancetype)_initWithCGImage:(CGImageRef)image color:(CIColor*)color {
```

Comment:
talked off line.
This is an internal initializer that's pretty much a wrapper around the actual designed path, without the use of this we would have to manually set the field :(

---
In reply to: [95705384](https://github.com/Microsoft/WinObjC/pull/1625#discussion_r95705384) [](ancestors = 95705384)